enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,608 @@
+[Unmanaged Subnet Annotation Migration Workflow](#unmanaged-subnet-annotation-migration-workflow).
+
+##### Reason #3: The CCM Doesn't Reconcile NLB Subnets Updates after creation.
+
+The CCM doesn't reconcile `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation updates to NLBs once it has been created. However, requiring a service
+deletion will effectuate these changes.
+
+See [CCM Doesn't Reconcile NLB Subnets Updates](#ccm-doesnt-reconcile-nlb-subnets-updates)
+for more details on why the CCM doesn't reconcile these updates.
+
+### Install Config Proposal (Day 1)
+
+TODO: Install time configuration proposal
+
+### Implementation Details/Notes/Constraints
+
+When an IngressController is created with
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+specified, the Ingress Operator will set the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation
+on the LoadBalancer-type Service.
+
+However, if `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+is updated on an existing IngressController, the Ingress Operator will NOT
+update `service.beta.kubernetes.io/aws-load-balancer-subnets` until the
+cluster admin deletes the LoadBalancer-type service as indicated by the
+`LoadBalancerProgressing` status (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+When the Ingress Operator recreates the LoadBalancer-type service, it will
+then configure `service.beta.kubernetes.io/aws-load-balancer-subnets` with
+the new `Subnets` value.
+
+The `status.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+will eventually reflect the _actual_ subnet value by mirroring the value of
+`service.beta.kubernetes.io/aws-load-balancer-subnets` on the service. There are
+two scenarios in which the `Subnets` status won't be equal to the `Subnets` spec:
+
+1. The cluster admin manually configured the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   annotation on the service.
+2. The cluster admin configured the IngressController's `Subnets` spec, but hasn't
+   effectuated the change by deleting the service (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+
+Note that this status usage is consistent with the approach in [LoadBalancer Allowed Source Ranges](/enhancements/ingress/lb-allowed-source-ranges.md),
+but diverges with [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md).
+The mutable publishing scope design set `status.endpointPublishingStrategy.loadBalancer.scope`
+to the initial value of `spec.endpointPublishingStrategy.loadBalancer.scope` so that it could
+detect when the spec for `Scope` changed. Therefore, `status.endpointPublishingStrategy.loadBalancer.scope`
+may not always reflect the actual scope of the load balancer. This proposal for `Subnets`
+perpetuates the pattern of ensuring that `status.endpointPublishingStrategy` reflects the
```

Comment:
Is it perpetuating the pattern of the previous sentence?  You lost me.

Reply:
I would give the pattern a good short name, like "triggered consistency", or something like that.

Reply:
This pattern I'm describing here doesn't really involve "triggered consistency".

This pattern is that: `status` == the effectuated value (i.e. the `service.beta.kubernetes.io/aws-load-balancer-subnets` in our case). 

This is a generic concept of how to keep our IngressController's status consistent, regardless if the effectuated value is triggered or delayed by having the cluster admin do something.

I think we have to spell this concept verbosely out here.

Reply:
I think you already gave it a name, the Load Balancer Allowed Subnets approach.  That works, since it is defined above.

Reply:
Sorry for the confusion. I explained in https://github.com/openshift/enhancements/pull/1595#discussion_r1591678677 too.

This isn't about the "Load Balancer Allowed Subnets approach" this is about maintaining consistency with the values in `status.endpointPublishingStrategy`. I see the confusion, and I'll refactor to make this clearer.